querybook/server/logic/user.py

```diff
@@ -144,6 +140,52 @@ def delete_user(uid, session=None):
     pass
 
 
+@with_session
+def create_or_update_user_group(user_group: UserGroup, commit=True, session=None):
+    group = get_user_by_name(user_group.name, session=session)
+    fields = {
+        "username": user_group.name,
+        "fullname": user_group.display_name,
+        "email": user_group.email,
+        "is_group": True,
+        "properties": {"description": user_group.description},
+    }
+
+    if not group:
+        # create a new group
+        group = User.create(
+            fields=fields,
+            commit=commit,
+            session=session,
+        )
+    else:
+        # update the group
+        group = User.update(
+            id=group.id,
+            fields=fields,
+            commit=commit,
+            session=session,
+        )
+
+    # get new member user ids by name
+    new_group_members = (
```

Comment:
i got a bit confused on "new" group members. It looks like they could also be not exist and already exist?

Have you considered instead of deleting all, only delete/create the deltas? it is more complicated but we would be able to use the created_at field to retain when the user joined the group